### PR TITLE
Refs #10970 - Removed katello_test_runner and updated glue tests

### PR DESCRIPTION
**Pull request by johnpmitsch**
new test.rake without test runners --> https://gist.github.com/johnpmitsch/9999259350b48056e3ec

### CODE REVIEW

File: app/models/katello/glue/pulp/pulp_content_unit.rb [hunk 1]
```diff
@@ -36,7 +36,7 @@ def in_repositories(repos)
       end
 
       def pulp_data(uuid)
-        unit_handler.find_by(:unit_id => uuid)
+        unit_handler.find_by_unit_id(uuid)
```
Comment (johnpmitsch) [inline app/models/katello/glue/pulp/pulp_content_unit.rb]: this is actually a runcible method

Comment (komidore64) [inline app/models/katello/glue/pulp/pulp_content_unit.rb]: I gotcha. So we don't want to switch to ActiveRecords new `find_by`. Got it.

Would it be worth looking into changing the syntax of Runcible at some point down the road? Or just leave it?

File: app/models/katello/glue/pulp/repo.rb [hunk 2]
```diff
@@ -241,7 +241,7 @@ def other_repos_with_same_content
         Repository.where(:content_id => self.content_id).pluck(:pulp_id) - [self.pulp_id]
       end
 
-      def rpm_ids
+      def package_ids
```
Comment (ehelms) [inline app/models/katello/glue/pulp/repo.rb]: Why this change?

File: app/models/katello/glue/pulp/repo.rb [hunk 3]
```diff
@@ -429,8 +429,8 @@ def puppet_module_ids
         Katello.pulp_server.extensions.repository.puppet_module_ids(self.pulp_id)
       end
 
-      def docker_image_ids
-        Katello.pulp_server.extensions.repository.docker_image_ids(self.pulp_id)
+      def docker_img_ids
```
Comment (ehelms) [inline app/models/katello/glue/pulp/repo.rb]: Why this name change?

File: test/glue/candlepin/consumer_test.rb [hunk 4]
```diff
@@ -11,7 +11,7 @@ class GlueCandlepinConsumerTestBase < ActiveSupport::TestCase
     @@dev_cv = nil
     @@dev_cve = nil
 
-    def self.before_suite
+    def setup
```
Comment (ehelms) [inline test/glue/candlepin/consumer_test.rb]: Does this still need the super call? If we are doing a `setup` call instead of a suite wrapper, I would think the class variables would not be needed ?

File: app/models/katello/glue/pulp/repo.rb (hunk 3 above)
Comment (johnpmitsch) [inline app/models/katello/glue/pulp/repo.rb]: both of this and the rpm_ids were returning the actual ids of rpms or docker images from the database instead of running the method, I guess this is a change in rails 4?

**Comment by ehelms** [comment]
Two other things:

1) There are before_suite methods in other files such as the `katello_test_helper` that would be rendered removable by this change.
2) This is the type of change that could potentially go straight into master.

**Comment by johnpmitsch** [comment]
@ehelms do you mean the all of the changes or just the removal of katello_test_runner would go straight to master?

File: app/models/katello/glue/pulp/repo.rb (hunk 3 above)
Comment (ehelms) [inline app/models/katello/glue/pulp/repo.rb]: Can you confirm or find that in the Rails 4 docs? This is less of a problem, but `rpm_ids` was used because that is specifically tied to RPMs as we were moving away from the generic 'package' in light of debian support down the road.

**Comment by ehelms** [comment]
Removal of the test runner (although most of this looks tied to that) -- typically any changes you can make to master directly will make getting the branch into master easier.

File: app/models/katello/glue/pulp/pulp_content_unit.rb (hunk 1 above)
Comment (johnpmitsch) [inline app/models/katello/glue/pulp/pulp_content_unit.rb]: @komidore64 I don't think its really affecting anything, IMO just leave it. @ehelms any opinion?

**Comment by johnpmitsch** [comment]
@ehelms ok, ya, I guess it would have to be the whole thing, because of the tests that have been updated not to use the test runner.

File: app/models/katello/glue/pulp/repo.rb (hunk 3 above)
Comment (johnpmitsch) [inline app/models/katello/glue/pulp/repo.rb]: @ehelms looking now

Comment (johnpmitsch) [inline app/models/katello/glue/pulp/repo.rb]: @ehelms http://edgeguides.rubyonrails.org/association_basics.html#has-many-association-reference scroll to 4.3.1.6 I am not sure if this was able to be overridden in rails 3 or just didn't exist.

**Comment by johnpmitsch** [comment]
opened up on master https://github.com/Katello/katello/pull/5503
